src/VAC/VectorAnimationComplex/TransformTool.cpp

```diff
@@ -1099,6 +1104,77 @@ void TransformTool::endDragAndDrop()
     dragAndDropping_ = false;
 }
 
+void TransformTool::setManualWidth(double newWidth, Time time)
+{
+    BoundingBox obb;
+    for (auto cell : cells_)
+    {
+        obb.unite(cell->outlineBoundingBox(time));
+    }
+
+    if (obb.isProper())
+    {
+        double delta = (newWidth - obb.width()) / 2;
+
+        if (abs(delta) > EPS)
+        {
+            hovered_ = LeftScale;
+            beginTransform(obb.xMin(), obb.yMid(), time);
+            continueTransform(obb.xMin() - delta, obb.yMid());
+            endTransform();
+
+            hovered_ = RightScale;
+            beginTransform(obb.xMax(), obb.yMid(), time);
+            continueTransform(obb.xMax() + delta, obb.yMid());
+            endTransform();
+            hovered_ = None;
+        }
+    }
+}
+
+void TransformTool::setManualHeight(double newHeight, Time time)
+{
+    BoundingBox obb;
+    for (auto cell : cells_)
+    {
+        obb.unite(cell->outlineBoundingBox(time));
+    }
+
+    if (obb.isProper())
+    {
+        double delta = (newHeight - obb.height()) / 2;
+
+        if (abs(delta) > EPS)
```

Comment:
I think Qt has some float equals function that handle EPS, prefer that if possible.

Reply:
Done